src/web/app/pages-help/instructor-help-page/instructor-help-page.component.ts

```diff
@@ -17,6 +18,8 @@ export class InstructorHelpPageComponent implements OnInit, AfterViewInit {
   readonly supportEmail: string = environment.supportEmail;
   searchTerm: String = '';
   key: String = '';
+  collapseStudentEditDetailsInChild: Subject<boolean> = new Subject<boolean>();
```

Comment:
I think this can just be
```suggestion
  collapseStudentEditDetailsInChild: boolean = false;
```

Reply:
I have tried this approach but it looks like it works only on the first trigger of the collapse event. If `collapseStudentEditDetailsInChild` in the parent (`instructor-help-page.component.ts`) is a boolean value and there is no 
`collapseStudentEditDetailsInChild: Subject<boolean>` in the child (`instructor-help-students-section.component.ts`), the child takes the value passed from the parent only once.

That is, with this configuration, in the following scenario the behavior is buggy:
1.  Upon the first trigger event, `collapseStudentEditDetailsInChild` is set to true in the parent and, thus, `isEditDetailsCollapsed` is set to true in the child. So, _Student-Edit-Details card_ is collapsed;
2. Then, suppose that user manually de-collapses this card and `isEditDetailsCollapsed` in the child becomes false;
3. Upon another trigger event, `collapseStudentEditDetailsInChild` is set to true in the parent but `isEditDetailsCollapsed` is not set to true in the child. So, _Student-Edit-Details card_ does not get collapsed though it should.

See:
![Screen-Recording-2020-04-11-at-1](https://user-images.githubusercontent.com/29358225/79042336-afe06700-7c18-11ea-8cb5-4c97cc0c6761.gif)

In contrast, in the current configuration, the child keeps listening to values from parent and the behavior is as expected.

Please correct me if I am missing anything. 
Thanks!

Reply:
I think the main reason here is that "you have two points of truth" (One `collapseStudentEditDetailsInChild` and one `isEditDetailsCollapsed`, there are not synced properly). It is possible to only have "single points of truth"? (i.e. All child components fetch their corresponding values from the source. If the parent wants to change the source, the child component will respond correspondingly. If any child component (not necessary the child component that use them) want to change the sources, they fire event and ask parents to change it.)

Reply:
Hi @xpdavid,

sorry for a delay, I had a busy last week with school assignments.
I have now attempted to have "single points of truth". Please see the last commits below. 

However, I still have one thing I am not quite sure about.

- As stated in the [Angular documentation](https://angular.io/guide/lifecycle-hooks#onchanges), the OnChanges lifecycle hook is triggered by Angular _only when the @Input property value changes_. This seems to explain why the @Input variables in the child components were not updating the values (as set by parent) when the @Input property value was not changing (this is the issue I addressed in the previous comment). 

- One suggested workaround is to ensure that Angular triggers the OnChanges hook by setting a new object reference with `Object.assign()` after modifying the property values. This is what I've done in the parent component.

Again, correct me if I am missing anything!
And thanks for your patience and guidance! :)